aya/src/programs/lirc_mode2.rs

```diff
@@ -78,10 +83,63 @@ impl LircMode2 {
 
         Ok(self.data.link(LircLink::new(prog_fd, lircdev_fd)))
     }
+
+    /// Query lirc device for attached programs
+    pub fn query<T: AsRawFd>(target_fd: T) -> Result<Vec<LircLink>, ProgramError> {
+        let mut prog_ids = vec![0u32; 64];
```

Comment:
Could you move the part that calls `bpf_prog_query` and populates the return ids to a shared  `pub(crate)` util in `aya::programs::query()`?

Reply:
done